### DISCUSSION TITLE
Local test of changes in library.json?

### DISCUSSION THREAD

**Discussion by o0shojo0o** [andrew-codechimp/HA-Battery-Notes#536] [Q&A] [answered]
Hi, 

Can I also change the library.json locally and test whether the device names I have determined match? 
If so, do I have to consider anything and then "arm" them?

**Comment by andrew-codechimp** [comment, marked as answer]
The library.json get's updated/overwritten every time you start HA but I've created myself a little test way of doing this.

Within your configuration.yaml add this
```
battery_notes:
  user_library: "library-dev.json"

```
Then create a new file in custom_components/battery_notes/data called library-dev.json with contents like below, replacing the details with your own. 
```
{
    "$schema": "../schema.json",
    "version": 1,
    "devices": [
        {
            "manufacturer": "Met.no",
            "model": "Forecast",
            "battery_type": "AA",
            "battery_quantity": 2
        }
    ]
}
```

Delete your existing battery note for the device you want to test, restart HA and anything in the user library will be matched first.

**Reply by o0shojo0o** [reply]
Super thank you for the answer, I have the fear that my last two "pulls" / issuses were faulty, so I can now test and better support :)

**Reply by andrew-codechimp** [reply]
And just noticed I don't even have to do a release to stop HACS overwriting the data folder, so you can leave your test data there even on an upgrade.
